zkvm/src/types.rs

```diff
@@ -301,11 +341,73 @@ impl Value {
 
 impl Expression {
     pub fn constant<S: Into<Scalar>>(a: S) -> Self {
-        let a: Scalar = a.into();
+        Expression::Constant(ScalarWitness::Scalar(a.into()))
+    }
+}
+
+impl Neg for Expression {
+    type Output = Expression;
+
+    fn neg(self) -> Expression {
+        match self {
+            Expression::Constant(a) => Expression::Constant(-a),
+            Expression::Terms(mut terms, assignment) => {
+                for (_, n) in terms.iter_mut() {
+                    *n = -*n;
```

Comment:
After thinking a bit more about `map` vs `iter_mut`, IMHO it's better to do `iter_mut()` to avoid reallocations.

Reply:
Maybe we could make it more readable?

- `terms.iter_mut().for_each(|x| x.1 = -x.1);`

- `for x in &mut a { x.1 = -x.1 }`

-  `terms.iter_mut().for_each(|(_, n)| *n = -*n)`

The below code should test that they work as expected:

```
fn main() {
    let mut a : Vec<(u64,i32)> = vec![(1,2),(2,3),(4,5),(6,7),(8,9)];
    a.iter_mut().for_each(|x| x.1 = -x.1);
    println!("{:?}",a);
    
    let mut a : Vec<(u64,i32)> = vec![(1,12),(2,13),(4,15),(6,17),(8,19)];
    for x in &mut a { x.1 = -x.1 };
    println!("{:?}",a);
    
    let mut a : Vec<(u64,i32)> = vec![(1,12),(2,13),(4,15),(6,17),(8,19)];
    a.iter_mut().for_each(|(_, n)| *n = -*n);
    println!("{:?}",a);
}
```

WDYT should we keep the same or change?

Reply:
We can leave `for (_, n) in terms.iter_mut() { *n = -*n }` imho.